src/diffusers/loaders/lora_pipeline.py

```diff
@@ -1125,7 +1136,13 @@ def load_lora_weights(
         # First, ensure that the checkpoint is a compatible one and can be successfully loaded.
         state_dict = self.lora_state_dict(pretrained_model_name_or_path_or_dict, **kwargs)
 
-        is_correct_format = all("lora" in key or "dora_scale" in key for key in state_dict.keys())
+        is_dora_scale_present = any("dora_scale" in k for k in state_dict)
+        if is_dora_scale_present:
```

Comment:
I see that we sometimes add this code to `lora_state_dict()` and sometimes on `load_lora_weights`. Is there any reason we do this differently?

Reply:
Sure.

It's because of the way `lora_state_dict()` and `load_lora_weights()` are used. 

`load_lora_weights()` is often the entry point to our users for loading a LoRA weight to perform inference. However, `lora_state_dict()` is also used during training, especially for resuming training from an existing checkpoint. Example:

https://github.com/huggingface/diffusers/blob/1287822973e87fc8739a608b7296fe1797b1b074/examples/dreambooth/train_dreambooth_lora_sdxl.py#L1264

LMK if that makes sense.

Reply:
In e08cf7442, I have tried to make this more consistent by moving all the "dora_scale" check related code to `lora_state_dict()`. 

LMK if that works for you, @yiyixuxu.